numpy/core/src/multiarray/methods.c

```diff
@@ -2577,8 +2577,9 @@ array_complex(PyArrayObject *self, PyObject *NPY_UNUSED(args))
     PyArrayObject *arr;
     PyArray_Descr *dtype;
     PyObject *c;
-    if (PyArray_SIZE(self) != 1) {
-        PyErr_SetString(PyExc_TypeError, "only length-1 arrays can "\
+
+    if (PyArray_NDIM(self) != 0) {
+        PyErr_SetString(PyExc_TypeError, "only rank-0 arrays can "\
                         "be converted to Python scalars");
```

Comment:
Responding to @seberg's question in the corresponding issue - this, and the other instance, should be:
```
if (PyArray_NDIM(self) != 0) {
    if (PyArray_SIZE(self) == 1) {
        if (DEPRECATE("Conversion of an array with one element to a scalar is deprecated, and will error in future") < 0) {
            return NULL;
        }
    }
    else {
        PyErr_SetString(PyExc_TypeError, "only rank-0 arrays can "
                        "be converted to Python scalars");
        return NULL;
    }
}
```

Reply:
Yeah, when I read the stuff, I thought of `PyArray_Result` first, thats why I thought it would be super complicated. This is fine, it is mostly about how many downstream pain credits we can/should spend on it.

Reply:
Probably worthwhile extracting a helper function so this code isn't duplicated.